helpers/frontpanel.cpp

```diff
@@ -190,6 +190,21 @@ namespace WPEFramework
             return s_instance;
         }
 
+        void CFrontPanel::Deinitialize(){
+
+            if(s_instance){
+                stop();
+                if(mFrontPanelHelper){
+                    delete mFrontPanelHelper;
+                    mFrontPanelHelper = nullptr;
+                }
+                delete s_instance;
+                s_instance = nullptr;
+                initDone = 0;
+                
+            }
+        }
+
```

Comment:
The problem here is initDone, which is a global variable defined in frontpanel.cpp. When I was redoing the void test cases to use EXPECT_EQUAL, I discovered a issue within my tests(as USE_DS was not set, and this fpIndicators was always 0), because since CFrontPanel never leaves scope, initDone, after the first run, is set to true, and never set to false, thus causing some values to become unsettable, since they're only set if initDone is set to false.

```
        CFrontPanel* CFrontPanel::instance()
        {
            if (!initDone)
            {
                if (!s_instance)
                    s_instance = new CFrontPanel;

#ifdef USE_DS
                try
                {
                    LOGINFO("Front panel init");
                    fpIndicators = device::FrontPanelConfig::getInstance().getIndicators();

                    for (uint i = 0; i < fpIndicators.size(); i++)
                    {
                        std::string IndicatorNameIarm = fpIndicators.at(i).getName();

                        auto it = std::find(m_lights.begin(), m_lights.end(), IndicatorNameIarm);
                        if (m_lights.end() == it)
                        {
                            m_lights.push_back(IndicatorNameIarm);
                        }
                    }

#if defined(HAS_API_POWERSTATE)
                    {
                        IARM_Bus_PWRMgr_GetPowerState_Param_t param;
                        IARM_Result_t res = IARM_Bus_Call(IARM_BUS_PWRMGR_NAME, IARM_BUS_PWRMGR_API_GetPowerState,
                            (void*)&param, sizeof(param));

                        if (res == IARM_RESULT_SUCCESS) {
                            if (param.curState == IARM_BUS_PWRMGR_POWERSTATE_ON)
                                powerStatus = true;
                        }
                    }
#endif
#ifdef CLOCK_BRIGHTNESS_ENABLED
                    clockBrightness =  device::FrontPanelTextDisplay::getInstance("Text").getTextBrightness();
                    device::FrontPanelTextDisplay::getInstance("Text").setTextBrightness(clockBrightness);
#endif
                    globalLedBrightness = device::FrontPanelIndicator::getInstance("Power").getBrightness();
                    LOGINFO("Power light brightness, %d, power status %d", globalLedBrightness, powerStatus);

                    for (uint i = 0; i < fpIndicators.size(); i++)
                    {
                        LOGWARN("Initializing light %s", fpIndicators.at(i).getName().c_str());
                        if (powerStatus)
                            device::FrontPanelIndicator::getInstance(fpIndicators.at(i).getName()).setBrightness(globalLedBrightness);

                        device::FrontPanelIndicator::getInstance(fpIndicators.at(i).getName()).setState(false);
                    }

                    if (powerStatus)
                        device::FrontPanelIndicator::getInstance("Power").setState(true);

                }
                catch (...)
                {
                    LOGERR("Exception Caught during [CFrontPanel::instance]\r\n");
                }
                initDone=1;
#endif
            }

            return s_instance;
        }



```
Without a way to set initDone to 0 (which is never done in the normal code), we will get segmentation faults any time anytime fpIndicators is accessed after the first test (because it gets set once, and can never be set again, and by the following tests, we've cleared out that instance that fpIndicators was using)

Reply:
Any suggestions on how to set initDone to 0 without adding a some kind of deinitialize function, since it doesn't get to set back to 0 anytime after it is set to 1. (Since we need to effectively re-set the values each time a new test is run, because we set the impl to a nullptr for the reference that is returned by getIndicators)